legend-engine-pure-code-compiled-core/src/main/resources/core/pure/router/preeval/preeval.pure

```diff
@@ -255,6 +255,22 @@ function <<access.private>> meta::pure::router::preeval::prevalInternal(item : A
   $newItem;
 }
 
+function <<access.private>> meta::pure::router::preeval::preevalAggregateFunction(f:FunctionDefinition<Any>[1], state : meta::pure::router::preeval::State[1], extensions:Extension[*]):PrevalWrapper<Any>[1]
+{
+  let fnArgNames = extractArgNames($f);
+  let fnOpenVariables = $f->openVariableValues()->putAll($state.inScopeVars->keyValues()->filter(kv | !$kv.first->in($fnArgNames))->newMap());
+
+  prevalInternal($f, ^$state(inScopeTypeParams = ^Map<String, GenericType>(), inScopeVars = $fnOpenVariables), $extensions);
+}
+
+function <<access.private>> meta::pure::router::preeval::extractArgNames(lambda:FunctionDefinition<Any>[1]):String[*]
+{
+  $lambda.classifierGenericType.typeArguments.rawType->map(rt | $rt->match([

```

Comment:
Is this function needed? (Does $lambda->functionType().parameters.name work?)
meta::pure::functions::meta::functionType_Function_1__FunctionType_1_